File: src/main/java/io/github/nikpivkin/lifetime/LifetimeTagType.java
package io.github.nikpivkin.lifetime;

import org.bukkit.persistence.PersistentDataAdapterContext;
import org.bukkit.persistence.PersistentDataType;
import org.jetbrains.annotations.NotNull;

public class LifetimeTagType implements PersistentDataType<Long, Lifetime> {

  public static final LifetimeTagType INSTANCE = new LifetimeTagType();

  @Override
  public @NotNull Class<Long> getPrimitiveType() {
    return Long.class;
  }

  @Override
  public @NotNull Class<Lifetime> getComplexType() {
    return Lifetime.class;
  }

  @Override
  public @NotNull Long toPrimitive(
      @NotNull Lifetime complex,
      @NotNull PersistentDataAdapterContext context
  ) {
    return complex.value();
  }

  @Override
  public @NotNull Lifetime fromPrimitive(@NotNull Long primitive,
      @NotNull PersistentDataAdapterContext context) {
    return new Lifetime(primitive);
  }
}

File: src/main/kotlin/io/github/nikpivkin/lifetime/LifetimeTagType.kt
package io.github.nikpivkin.lifetime

import org.bukkit.persistence.PersistentDataAdapterContext
import org.bukkit.persistence.PersistentDataType

object LifetimeTagType : PersistentDataType<Long, Lifetime> {
    override fun getPrimitiveType() = Long::class.java

    override fun getComplexType() = Lifetime::class.java

    override fun fromPrimitive(
        primitive: Long,
        context: PersistentDataAdapterContext
    ) = Lifetime(primitive)

    override fun toPrimitive(
        complex: Lifetime,
        context: PersistentDataAdapterContext
    ) = complex.value()
}
